### PR TITLE
feat: update about b&t process component

### PR DESCRIPTION
**Pull request by andyesp**

### CODE REVIEW

**Review by ncomerci — APPROVED** [review]

File: src/hooks/useTenderProposals.ts [hunk 1]
```diff
@@ -17,5 +17,7 @@ export function useTenderProposals(
     tenderProposals: proposals,
     isLoadingTenderProposals: isLoadingProposals,
     hasTenderProcessStarted: !!proposals?.data && hasTenderProcessStarted(proposals?.data),
+    hasTenderProcessFinished: !!proposals?.data && hasTenderProcessFinished(proposals?.data),
```
Comment (1emu) [inline src/hooks/useTenderProposals.ts:20]: `hasTenderProcessFinished` & `hasTenderProcessStarted` could allow for null|undefined params and return false for those cases, so we can skip checking when calling them

File: src/pages/proposal.tsx [hunk 2]
```diff
@@ -371,9 +373,9 @@ export default function ProposalPage() {
               {proposal?.type === ProposalType.POI && <ProposalFooterPoi configuration={proposal.configuration} />}
               {showTenderProposals && <TenderProposals proposals={tenderProposals.data} />}
               {showBidProposals && <BidProposals proposals={bidProposals.data} />}
-              {proposal && isBiddingAndTenderingProposal(proposal.type) && (
-                <BiddingAndTenderingProcess proposal={proposal} tenderProposalsTotal={tenderProposals?.total} />
-              )}
+              {proposal && proposal.type === ProposalType.Pitch && <AboutPitchProcess proposal={proposal} />}
```
Comment (1emu) [inline src/pages/proposal.tsx:376]: How about something like

`proposal & <BiddingAndTenderingProcess proposal={proposal} />`

and 

```typescript
export default function BiddingAndTenderingProcess({ proposal }: Props) {
  switch (proposal.type) {
    case ProposalType.Pitch:
      return <AboutPitchProcess proposal={proposal} />
    case ProposalType.Tender:
      return <AboutTenderProcess proposal={proposal} />
    case ProposalType.Bid:
      return <AboutBidProcess proposal={proposal} />
    default:
      return null
  }
}
```

Just so proposal.tsx doesn't get too much bigger

**Review by ncomerci — APPROVED** [review]

**Review by 1emu — APPROVED** [review]

File: src/components/Proposal/View/AboutPitchProcess.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        !!tenderProposalsTotal && tenderProposalsTotal > 0
+          ? 'page.proposal_bidding_tendering.tender_proposal_submitted'
+          : 'page.proposal_bidding_tendering.tender_proposal_pending',
+    }
+  }
+
+  return { status: ProcessStatus.Default, statusText: 'page.proposal_bidding_tendering.tender_proposal_requires' }
+}
+
+const getOpenForBidsConfig = (hasBid: boolean, hasBidProposals: boolean) => {
```
[the remark this answers is not in the document]
Comment (andyesp) [reply src/components/Proposal/View/AboutPitchProcess.tsx]: Not anymore

File: src/components/Proposal/View/AboutPitchProcess.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  }
+
+  if (hasBidProposals) {
+    return { status: ProcessStatus.Passed, statusText: 'page.proposal_bidding_tendering.open_for_bids_inbound' }
+  }
+
+  return { status: ProcessStatus.Default, statusText: 'page.proposal_bidding_tendering.open_for_bids_requires' }
+}
+
+const getProjectAssignationConfig = (hasWinnerBid: boolean, hasRejectedBid: boolean) => {
```
[the remark this answers is not in the document]
Comment (andyesp) [reply src/components/Proposal/View/AboutPitchProcess.tsx]: Only in 2/3 components. I'll wait to reuse it.

**Pull request merged**